fireant/slicer/transformers/datatables.py

```diff
@@ -274,16 +270,34 @@ def _render_column_level(self, metric_column, display_schema):
         }
 
     def _recurse_dimensions(self, dataframe, dimensions, metrics, reference=None):
+        if dataframe.empty:
+            return []
+
         if reference or not dimensions:
             return super(DataTablesColumnIndexTransformer, self)._recurse_dimensions(dataframe, dimensions, metrics,
                                                                                      reference)
 
+        data = []
+
         if 'display_field' in dimensions[0][1]:
-            return [(key, dict(self._recurse_dimensions(dataframe[:, key, display], dimensions[1:], metrics)))
-                    for key, display in zip(*dataframe.index.levels[1:3])]
+            levels = zip(*dataframe.index.levels[1:3])
+            format_key = lambda level: level[0]
+            generate_dataframe = lambda level: dataframe[:, level[0], level[1]]
+        else:
+            levels = dataframe.index.levels[1]
+            format_key = lambda level: str(level)
```

Comment:
This raises a codacy issue. You could just set format_key = str to avoid the lambda.